gremlin-console/src/main/groovy/org/apache/tinkerpop/gremlin/console/GremlinGroovysh.groovy

```diff
@@ -140,4 +171,48 @@ class GremlinGroovysh extends Groovysh {
 
         maybeRecordResult(result)
     }
+
+    private Object evaluateWithStoredBoundVars(String importsSpec, List<String> current) {
+        Object result
+        String variableBlocks = null
+        // To make groovysh behave more like an interpreter, we need to retrieve all bound
+        // vars at the end of script execution, and then update them into the groovysh Binding context.
+        Set<String> boundVars = ScriptVariableAnalyzer.getBoundVars(importsSpec + Parser.NEWLINE + current.join(Parser.NEWLINE), interp.classLoader)
+        if (boundVars) {
+            variableBlocks = "$COLLECTED_BOUND_VARS_MAP_VARNAME = new HashMap();"
+            boundVars.each({ String varname ->
+                // bound vars can be in global or some local scope.
+                // We discard locally scoped vars by ignoring MissingPropertyException
+                variableBlocks += """
+try {$COLLECTED_BOUND_VARS_MAP_VARNAME[\"$varname\"] = $varname;
+} catch (MissingPropertyException e){}"""
+            })
+        }
+        // Evaluate the current buffer w/imports and dummy statement
+        List<String> buff
+        if (variableBlocks) {
+            buff = [importsSpec] + ['try {', 'true'] + current + ['} finally {' + variableBlocks + '}']
+        } else {
+            buff = [importsSpec] + ['true'] + current
+        }
+        setLastResult(result = interp.evaluate(buff))
```

Comment:
I'm a bit concerned here that we just do setLastResult() to whatever is in the buffer here. Notice how for remote console we normally do this with something like `mediator.currentRemote().submit([script])`. Have you checked to see if this is a problem?

Reply:
Confirmed that multiline queries are fixed for both interpreter modes